### ISSUE TITLE
Importing browser data on Android

### ISSUE THREAD

**Issue by jonalmeida** [mozilla/platform-tilt#7]
Browsing information like history, bookmarked sites, and cookies isn’t accessible to third-party browsers on Android. This data is kept within a web browser application’s [data directory](https://developer.android.com/training/data-storage/app-specific#internal), which isn’t directly accessible to third-party browsers, and there’s no API or [ContentProvider](https://developer.android.com/guide/topics/providers/content-provider-basics) to enable it to be imported. While this is sensitive data, similar import functionality is possible on all major desktop platforms, and Android is able to mediate access to other sensitive data with user consent. Not being able to import data creates significant friction to change from Chrome - a user should be allowed to bring their data with them to another browser.

**Comment by iamgitcat** [comment]
Than why I can not bring Firefox collection bookmarks from android to desktop? Also Chrome can not import Firefox data either.

**Comment by Anduin1357** [comment]
@jonalmeida
Isn't it already possible to sync brower data from Chrome Android to Desktop Chrome to import to Firefox Desktop and finally to sync to Firefox Android?

Given that getting access to desktop versions of Chrome and Firefox is trivial, this isn't a big issue.

@iamgitcat
You can already access Firefox bookmarks made on any device, provided that they are synced. You may need to go up a directory level to see the other bookmark folders.

**Comment by iamgitcat** [comment]
> You can already access Firefox bookmarks made on any device, provided that they are synced. You may need to go up a directory level to see the other bookmark folders.

No.. Collections in Firefox for Android. On a homepage, I love this feature as temporary bookmarks. They could not be synced, not even with another Android.

**Comment by Anduin1357** [comment]
You can share it and push them to your browser, or as text to any of the common share options on android.

Besides, you may want to consider using an extension that better deals with saving tabs such as OneTab, now that desktop extensions on Android is possible.

**Comment by kcarrandale** [comment]
> Given that getting access to desktop versions of Chrome and Firefox is trivial, this isn't a big issue.

Tens of millions of people in the US only access the Internet by phone, and while I don't have specific numbers I'm certain that's even more common worldwide. Expecting users to have access to a desktop version of Chrome and Firefox to move bookmarks and other data is unreasonable (and also expects a decent level of technical literacy due to the extra steps).

**Comment by willfs84** [comment]
Browsers on Android should be allowed to ask users if they would like to import their history, bookmarks etc. from another browser when they run for the first time. On Windows, Firefox is able to do this, this is what it looks like. 

![Image](https://github.com/user-attachments/assets/4c8bcd8a-0245-48dc-8663-4d5641d73e33)

**Comment by qwirl** [comment]
https://github.com/mozilla/platform-tilt/issues/7#issuecomment-1901945006
>Isn't it already possible to sync brower data from Chrome Android to Desktop Chrome to import to Firefox >Desktop and finally to sync to Firefox Android?
>
>Given that getting access to desktop versions of Chrome and Firefox is trivial, this isn't a big issue.

I refuse to use an online tool like Firefox sync, for a job that is an offline task, and being forced to use Firefox Sync is a privacy non-starter.

Firefox claims to fight for user Privacy, but it's all the chromium based browsers (+ others) that are having a bookmarks import/export feature, and it's easy to transfer bookmarks between those browsers on Android.

Currently, if you are on Android, and want to export bookmarks on Android, without reach of another device: look at the sun shining, it's not happening.

Last time I managed to export Firefox bookmarks on Android, was by:

- Having a rooted device.
- Finding the browser.db file under /data/data/org.mozilla..something/xxxxxxxx.default/files/browser.db, that contained the bookmarks that I wanted to export.
- Placed the browser.db file to a regular user accessible location.
- Opening the browser.db in Privacy Browser.
- Importing the bookmarks to Privacy Browser.
- Exporting the bookmarks from Privacy Browser as an HTML file.

Does this really have to be necessary?

That worked once, and neither do I know how or why it worked, but maybe because Privacy Browser used a database that was a clone of the Firefox database.
They where named the same, so I tried, with success.

Now my issue is that Firefox don't do Bookmarks export, and that makes me want to consider my options, as that is a very essential feature.

I saw once one talk about it being hard to implement, and now I'm not a developer, but I have programmed before, so why is it a struggle to make database queries, and save it to a HTML file?

>You can already access Firefox bookmarks made on any device, provided that they are synced. You may need to go up a directory level to see the other bookmark folders.

You are ignoring the fact, that Firefox Sync isn't a great way to provide privacy, compared to a HTML bookmarks export, and you are assuming that the any person in question have several devices.
Assuming how people shell live their life is evil, which includes how many devices they should have, and I do not expect that of Firefox, but tell me if I'm wrong.


https://github.com/mozilla/platform-tilt/issues/7#issuecomment-1901988658
>You can share it and push them to your browser, or as text to any of the common share options on android.
>
>Besides, you may want to consider using an extension that better deals with saving tabs such as OneTab, now >that desktop extensions on Android is possible.

Making users share 1000+ bookmarks one by one, or open 1000+ bookmarks in OneTab is ignorant and inconsiderate. As it is right now, only one bookmark is shared at a time, even if several/all bookmarks are shared at the same time.


If Mozilla wants to make standardized approaches, then start by making an import/export that is already present in other browsers (put cars on the road), and then you can strive for holistic import/export of data like history, bookmarked sites, and cookies all together.
